### ISSUE TITLE
`arrange_spans` throws with an integer overflow for very large huxtables

### ISSUE THREAD

**Issue by kpagacz** [hughjonesd/huxtable#256]
TL;DR: `arrange_spans` overflows the integer limit, here's the fix: https://github.com/hughjonesd/huxtable/pull/257

## Description

The internal function `arrange_spans` throws an integer overflow exception for very large huxtables (tables large enough that the square of its largest diameter exceeds the limits of `int32_t`). 

## Minimal reproducible example

`arrange_spans` is not part of the public API, but it is a part of many of `huxtables`'s mechanism (including subsetting, copying, and printing to PDF)

```{r}
h1 <- huxtable::huxtable(a = rep(1, 2 ** 17), b = rep(2, 2 ** 17))
h2 <-huxtable::huxtable(a = rep(1, 2 ** 17), b = rep(2, 2 ** 17))

# The below throws the error
huxtable:::arange_spans(h1, h2)
```

The offending piece of code is located here: https://github.com/hughjonesd/huxtable/blob/cbb3a7fa4c22aa9f8bf32c23168bbd0d6ae64773/R/subset-extract-internal.R#L187

If any of the values in `dc$display_row * stride` exceed the upper limit of `int32_t` the error is thrown. What is worth highlighting is that this multiplication is not essential for the rest of the code to work. The values calculated in `merge_sets` are only used to identify the cell that is being displayed.

## Solution

I took the liberty of creating a small PR (https://github.com/hughjonesd/huxtable/pull/257) that prevents this overflow. The proposed solution changes the displayed cell identifier from a single value to a pair of coordinates (row and column) in `old_ht`. The following comparisons are also refactored to adapt to this change.

This has the drawback of `merge_sets` taking twice as much memory as before, but it prevents the integer overflow error. Another solution would be to chunk the computations and create `merge_sets` on chunks of `old_ht` and process further in chunks. I opted not to implement this solution because:
* the logic to detect the chunks is not trivial (a chunk needs to contain all its merged cells)
* if a merged cell spans all the columns or all the rows, then this solution would degrade to the original implementation and would overflow anyway (due to the need of any chunk to contain all the merged cells that start in it)

**Comment by hughjonesd** [comment]
I'm fascinated... what's the use case for creating a million-row display table?

**Comment by kpagacz** [comment]
Some of the listings `JNJ` is working with are very large (clinical data, sometimes there are that many rows in the tables). The issue might go away because some chunking mechanisms are being introduced to the R scripts that produce the outputs for the clinical statisticians. Still, it'd be nice to avoid this overflow anyway as the process is being introduced slowly and in some cases it's particularly hard to implement.

**Comment by hughjonesd** [comment]
I lean to the view that if you are turning 65000 rows of data into a formatted table, then you may be doing something silly. Nobody can read that much, so why not only format the bits people will read?

**Comment by hughjonesd** [comment]
Have you tried the slightly simpler fix:

```r
  merge_sets <- as.double(dc$display_row) * stride + dc$display_col
```

This seems to work for me, from a quick informal test.

**Comment by hughjonesd** [comment]
Please could you try github master and confirm if this fixes your bug? Thanks.
